src/identity.cc

```diff
@@ -592,6 +718,17 @@ TRITONBACKEND_ModelInstanceExecute(
        " requests")
           .c_str());
 
+  // Delay if requested...
+  if (model_state->ExecDelay() > 0) {
+    int multiplier = 1;
+    if (model_state->Multiplier() > 0) {
+      multiplier *= instance_state->DeviceId();
```

Comment:
The legacy implementation is not setting multiplier based on device id, instead, it is based on the instance id because device id will always be -1 for CPU instance and that can't be used to set different delays for different instances

Reply:
I used DeviceId since we set device_id to instance_id in the ModelInstanceState constructor

Reply:
@deadeyegoodwin I think you had done this. Was there a reason for setting instance_id to device_id?

Reply:
I didn't change this code from "instance id" to "device id", did I?

Reply:
> 
> 
> I used DeviceId since we set device_id to instance_id in the ModelInstanceState constructor

That's not right. Can you point to that code.

Reply:
I think I misunderstood earlier. 
https://github.com/triton-inference-server/identity_backend/blob/main/src/identity.cc#L314 is the constructor

https://github.com/triton-inference-server/identity_backend/blob/main/src/identity.cc#L308-L310 is the call that initialized device_id_ but although instance_id is the name of the variable it is set using TRITONBACKEND_ModelInstanceDeviceId and is likely the device id and not instance id (correct?)

Reply:
Model instances have unique names but they are not assigned unique ids. The backend API lists everything you can know about an instance: https://github.com/triton-inference-server/core/blob/main/include/triton/core/tritonbackend.h#L787

What is "Multiplier" used for. It is strange to multiply it by either instance id (if there was such a thing), or device id.

Reply:
It is used to make different instances have different execution delay so the later requests are finished early in the test, so that we can check if order is preserved in scheduler even if that happens

Reply:
This carries over from the legacy custom backend. we have an instance_wise_delay_multiplier which is what Multiplier here refers to https://github.com/triton-inference-server/identity_backend/blob/hemantj-identity/src/identity.cc#L328-L339

The corresponding section to the custom backend is here: https://github.com/triton-inference-server/server/blob/master/src/test/custom/identity/identity.cc#L124-L143
There is logic in the legacy custom backend here: https://github.com/triton-inference-server/server/blob/d71a2d8c70fe0b58ebf187b079d0ccd02b327a52/src/backends/custom/custom_backend.cc#L122-L134 that creates the name using the "instance id"

It is used inside L0_batcher here: https://github.com/triton-inference-server/server/blob/d71a2d8c70fe0b58ebf187b079d0ccd02b327a52/qa/L0_batcher/test.sh#L422

Reply:
OK, inside identity model state you will just have to keep an instance count that you can update each time ModelInstanceInitialize is called and then assign that as the unique instance id.

Reply:
Done